### ISSUE TITLE
allow rename variables

### ISSUE THREAD

**Issue by Norlandz** [mark-wiemer-org/ahkpp#422]
**Is your feature request related to a problem? Please describe.**
I want to **rename** a variable -> press `F2` -> error: `You cannot rename this element`
![image](https://github.com/mark-wiemer-org/ahkpp/assets/43581880/49d3d2ec-0eb5-4615-83f0-2a082fdc02a7)

**Describe the solution you'd like**
allow rename variables

**Additional Contents**
- Im using AHK v1
- when I ctrl click the var, it does link me to the var declaration, 
  so it should able to know how to link all the variables.
- Idk why seems AutoHotkey v2 Language Support https://marketplace.visualstudio.com/items?itemName=thqby.vscode-autohotkey2-lsp
  is able to do the rename?
  ![](https://raw.githubusercontent.com/thqby/vscode-autohotkey2-lsp/d16848c3e679edf4b0f1d80202e292fa4d5e0a7f/pic/rename.gif)
- (or did i config anything wrong?)

**Comment by mark-wiemer** [comment]
I don't think you configured anything wrong. Renaming variables is not currently supported, but is a great suggestion for a future release. I'll keep this on the backlog, thanks for reporting it :)

**Comment by marium0505** [comment]
Seems like this can be closed (PS: have disabled AutoHotkey v2 Language Support just in case the feature came from that)? 
![image](https://github.com/user-attachments/assets/346268fb-4c91-4c55-8ee7-bd6b1c294a8b)

**Comment by mark-wiemer** [comment]
![image](https://github.com/user-attachments/assets/9ce5ab36-681b-4dd5-a15b-d1c8b0b5eae5)

Are you still using AHK v1? I'm still getting the error when testing locally

**Comment by marium0505** [comment]
Ahh, sorry, you’re right, I’m on v2. , should have read the OP more
carefully.
